### PR TITLE
Added relu, sigmoid, leaky_relu, log_sigmoid, silu, swish, softmax

### LINKED ISSUE #15794: relu

### LINKED ISSUE #15893: sigmoid

### LINKED ISSUE #15894: leaky_relu

### LINKED ISSUE #15897: log_sigmoid

### LINKED ISSUE #15899: silu

### LINKED ISSUE #15900: swish

### LINKED ISSUE #15901: softmax

### PR DESCRIPTION
**Pull request by KSANTHOSH200**
Close #15794
Close #15893 
Close #15894
Close #15897
Close #15899
Close #15900
Close #15901

### CODE REVIEW

File: ivy/functional/frontends/paddle/nn/functional/activation.py [hunk 1]
```diff
@@ -1,5 +1,46 @@
...
 
-tanh = paddle_tanh
+@with_supported_dtypes({"2.4.2 and below": ("float32", "float64")}, "paddle")
+@to_ivy_arrays_and_back
+def sigmoid(x, name=None):
+    return ivy.sigmoid(x)
+
+@with_supported_dtypes({"2.4.2 and below": ("float32", "float64")}, "paddle")
+@to_ivy_arrays_and_back
+def leaky_relu(x, alpha=0.2, name=None):
```
Comment (hmahmood24) [inline ivy/functional/frontends/paddle/nn/functional/activation.py:24]: The second parameter is `negative_slope` not `alpha` in the [official docs](https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/nn/functional/leaky_relu_en.html), with a default value of `0.01`

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_activation.py [hunk 2]
```diff
@@ -1,3 +1,184 @@
...
+    test_flags,
+):
+    input_dtype, x = dtype_and_x
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        alpha=1e-2,
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_activation.py:80]: Would need to change the parameter name here as well

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_activation.py [hunk 3]
```diff
@@ -1,3 +1,184 @@
...
+    )
+
+# silu
+@handle_frontend_test(
+    fn_tree="paddle.tensor.math.log_silu",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"),
+    ),
+)
+def test_paddle_log_silu(
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_activation.py:116]: Forgot to change the name of the test here perhaps?

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_activation.py [hunk 4]
```diff
@@ -1,3 +1,184 @@
...
+    )
+
+# swish
+@handle_frontend_test(
+    fn_tree="paddle.tensor.math.log_swish",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"),
+    ),
+)
+def test_paddle_log_silu(
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_activation.py:141]: Here as well

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_activation.py [hunk 5]
```diff
@@ -1,3 +1,184 @@
...
+
+# softmax
+@handle_frontend_test(
+    fn_tree="paddle.tensor.math.log_softmax",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"),
+    ),
+    axis=st.integers(),
+)
+def test_paddle_log_silu(
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_activation.py:167]: Here too

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_activation.py [hunk 6]
```diff
@@ -1,3 +1,184 @@
 # global
+from hypothesis import strategies as st
 
 # local
+import ivy_tests.test_ivy.helpers as helpers
+from ivy_tests.test_ivy.helpers import handle_frontend_test
+
+# relu
+@handle_frontend_test(
+    fn_tree="paddle.tensor.math.relu",
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_activation.py:10]: The path is incorrect in all of the tests. Use path `paddle.nn.functional.<func_name>` here for example `paddle.nn.functional.relu`. Kindly make this change in all of the tests in this file

File: ivy/functional/frontends/paddle/nn/functional/activation.py [hunk 7]
```diff
@@ -1,5 +1,46 @@
...
+
+@with_supported_dtypes({"2.4.2 and below": ("float32", "float64")}, "paddle")
+@to_ivy_arrays_and_back
+def swish(x, name=None):
+    return ivy.vecdot(x, ivy.sigmoid(x))
+
+@with_supported_dtypes({"2.4.2 and below": ("float32", "float64")}, "paddle")
+@to_ivy_arrays_and_back
+def softmax(x, axis=-1, name=None):
+    axis= None if axis=-1 else axis
```
Comment (hmahmood24) [inline ivy/functional/frontends/paddle/nn/functional/activation.py:45]: Syntax error. The program will break here. Kindly fix this

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @KSANTHOSH200, thank you for making this PR. I have some points I want to make:

1. I have left multiple comments on the PR for you to address. Kindly make sure you make all those changes.
2. There are a lot of lint/formatting issues with this PR. I am guessing you haven't installed `pre-commit` but I highly recommend doing so to fix [these](https://github.com/unifyai/ivy/actions/runs/5087986076/jobs/9143883375?pr=15895) linting issues.
3. You haven't tested your functions locally using `pytest` because the program would have exited without running the test because of some syntactical and coding mistakes I pointed out in the review. You can also verify that the CI didn't run any tests from [here](https://github.com/unifyai/ivy/actions/runs/5087986054/jobs/9143882950?pr=15895#step:3:135). Kindly **always** make sure you run your tests locally first and ensure that they pass for all backends before opening a PR.
4. Please do not include multiple functions from ToDo list into a single PR. The policy is to create separate PRs for separate functions. Adding so many functions to the same PR gets difficult not only for the maintainer but also for the contributor to test, debug, and get merged. Kindly make sure you follow the 1 function 1 PR approach after this.

Kindly incorporate all these changes/points and re-request a review from me once you've verified that your functions are passing for all the backends. Let me know if you need any help or didn't understand anything. Thanks and good luck! 😃
